src/Plugin/resource/Field/ResourceFieldEntity.php

```diff
@@ -0,0 +1,237 @@
+   * property or sub property (e.g. $wrapper->field_reference->label()).
+   *
+   * @var bool
+   */
+  protected $wrapperMethodOnEntity = FALSE;
+
+  /**
+   * If the property is a field, set the column that would be used in queries.
+   * For example, the default column for a text field would be "value". Defaults
+   * to the first column returned by field_info_field(), otherwise FALSE.
+   *
+   * @var string
+   */
+  protected $column;
+
+  /**
+   * Array of image styles to apply to this resource field maps to an image
+   * field.
+   *
+   * @var array
+   */
+  protected $imageStyles = array();
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(array $fields, ResourceFieldInterface $decorated = NULL) {
+    // Create the current object.
+    $resource_field = new static($fields);
+
+    // Set the basic object to the decorated property.
+    $resource_field->decorate($decorated ? $decorated : ResourceField::create($fields));
+
+    // Add the default specifics for the current object.
+    $resource_field->addDefaults();
+
+    return $resource_field;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function decorate(ResourceFieldInterface $decorated) {
+    $this->decorated = $decorated;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSubProperty() {
+    return $this->subProperty;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setSubProperty($subProperty) {
+    $this->subProperty = $subProperty;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormatter() {
+    return $this->formatter;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setFormatter($formatter) {
+    $this->formatter = $formatter;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getWrapperMethod() {
+    return $this->wrapperMethod;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setWrapperMethod($wrapperMethod) {
+    $this->wrapperMethod = $wrapperMethod;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isWrapperMethodOnEntity() {
+    return $this->wrapperMethodOnEntity;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setWrapperMethodOnEntity($wrapperMethodOnEntity) {
+    $this->wrapperMethodOnEntity = $wrapperMethodOnEntity;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getColumn() {
+    return $this->column;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setColumn($column) {
+    $this->column = $column;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getImageStyles() {
+    return $this->imageStyles;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setImageStyles($imageStyles) {
+    $this->imageStyles = $imageStyles;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addDefaults() {
+    // Almost all the defaults come are applied by the object's property
+    // defaults.
+
+    // Set the defaults from the decorated.
+    $this->setResource($this->decorated->getResource());
```

Comment:
This sets the resource from `ResourceFieldBase` to be the resource for `ResourceFieldEntity`. This is just a way to simulate inheritance without having to use a complete decorator.

We want to avoid inheritance because we could get in these situations were:
  - A contrib module adds a class `AcmeResourceFieldEntity` that inherits from `ResourceFieldEntity`.
  - In parallel we inherit from `ResourceFieldEntity` to create `CachedResourceFieldEntity`.

Now there are 2 different inheritance branches, and no `AcmeCachedResourceFieldEntity` is possible without a rewrite.